src/main.rs

```diff
@@ -1,60 +1,254 @@
+    Ok(chain.chain)
+}
+
+use clap::{Parser, Subcommand};
+
+#[derive(Parser)]
+#[command(name = "bitcoin-rpc-cli")]
+#[command(version = "0.1.0")]
+#[command(about = "A CLI for the Bitcoin RPC API", long_about = None)]
+struct Cli {
+    #[command(subcommand)]
+    command: Option<Commands>,
+}
+
+#[derive(Subcommand)]
+enum Commands {
+    #[command(about = "Get the current chain")]
+    Chain,
+    #[command(about = "Get the it took to mine a block")]
+    TimeToMine {
+        #[arg(required = true, help = "(numeric, required) The height index")]
+        block_height: u64,
+    },
+    #[command(about = "Get the number of transactions in a block")]
+    NumberOfTransactions {
+        #[arg(required = true, help = "(numeric, required) The height index")]
+        block_height: u64,
+    },
+    #[command(about = "Guess how long until next block is mined")]
+    NextBlock,
+}
+
+// QUESTION: is this the best way to make error handling happen in a single place?
+// if command returns an error then return it to whatever called `call_command`
+fn call_command(command: Commands) -> std::result::Result<(), Box<dyn Error>> {
+    match command {
+        Commands::Chain => {
+            let chain = get_chain()?;
+            println!("{}", chain);
+        }
+        Commands::TimeToMine { block_height } => {
+            let time = time_to_mine(block_height)?;
+            println!("{}s, {}min", time.num_seconds(), time.num_minutes());
+        }
+        Commands::NumberOfTransactions { block_height } => {
+            let num = number_of_transactions(block_height)?;
+            println!("{} transactions", num);
+        }
+        Commands::NextBlock => {
+            println!("Next block will be mined in: ");
+            let time = guess_time_to_mine_next_block()?;
+            println!("{}s, {}min", time.num_seconds(), time.num_minutes());
+        }
+    };
+    Ok(())
 }
 
 fn main() {
-    // you can use rpc_client here as if it was a global variable
-    // println!("{:?}", res);
-    const TIMEOUT_UTXO_SET_SCANS: time::Duration = time::Duration::from_secs(60 * 8); // 8 minutes
-    dotenv::dotenv().ok();
-        let rpc_url: String = env::var("BITCOIN_RPC_URL").expect("BITCOIN_RPC_URL must be set");
-        let rpc_user: String = env::var("BITCOIN_RPC_USER").expect("BITCOIN_RPC_USER must be set");
-        let rpc_password: String =
-            env::var("BITCOIN_RPC_PASSWORD").expect("BITCOIN_RPC_PASSWORD must be set");
+    let cli: Cli = Cli::parse();
+
+    if let Err(e) = match cli.command {
+        Some(cmd) => call_command(cmd),
+        None => {
+            eprintln!("No command provided");
+            Ok(())
+        }
+    } {
+        eprintln!("Error: {}", e);
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_get_chain() {
+        // QUESTION: how to test just that this returns instance of Network? Or is that redundant because of type safety. What would be a better test if so?
```

Comment:
yeah you have a guarantee that as long as you have an OK result, you get an instance of Network due to type safety

If you want, you could do something like this using destructuring:
```rust
if let Network = chain {
  assert!(true)
} else {
  panic!("Unexpected network type")
}
```